### PR TITLE
Export all - Projects view - projects endpoints (unit tests) [SCI-2732]

### PR DESCRIPTION
**Pull request by mz3944**
[SCI-2732](https://biosistemika.atlassian.net/browse/SCI-2732)

### CODE REVIEW

**Comment by mz3944** [comment]
Some tests didn't pass, but they worked locally. But that exposed a misplaced test context, so that's good.

File: spec/services/projects_overview_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+                     archived: false, created_at: time.advance(hours: 4)
+  end
+  (7..PROJECTS_CNT).each do |i|
+    let!("project_#{i}") do
+      create :project, name: "test project #{(64 + i).chr}",
+                       visibility: 1,
+                       team: team, archived: i % 2,
+                       created_at: time.advance(hours: 6, minutes: i)
+    end
+  end
```
[the remark this answers is not in the document]
Comment (mz3944) [reply spec/services/projects_overview_service_spec.rb]: It's a test, I would leave it as it is.

File: spec/services/projects_overview_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+        expect(projects1.length).to eq 10
+      end
+
+      context 'with request parameters { start: 15 }' do
+        let(:params) { super().merge(start: 15) }
+
+        it 'returns archived projects, sorted by ascending archivation ' \
+           'attribute (active first), offset by 15, paginated by 10' do
+          projects = projects_overview.projects_datatable(params)
+          expect(projects.length).to eq 3
```
[the remark this answers is not in the document]
Comment (mz3944) [reply spec/services/projects_overview_service_spec.rb]: I did so only for the cases where the first six, specifically tailored projects would be present in the result. Hence, I would leave it as is.

File: spec/services/projects_overview_service_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mz3944) [reply spec/services/projects_overview_service_spec.rb]: Actually, we could write a factory for this... another day perhaps.

**Comment by jbargu** [comment]
Looks good, merge away!

**Pull request merged**